cli/config.go

```diff
@@ -4,65 +4,211 @@
 package cli
 
 import (
-	"fmt"
-	"log"
+	"io"
 	"os"
+	"reflect"
+	"strconv"
 
 	"github.com/mainflux/mainflux/pkg/errors"
+	mfxsdk "github.com/mainflux/mainflux/pkg/sdk/go"
 	"github.com/pelletier/go-toml"
+	"github.com/spf13/cobra"
 )
 
+type remotes struct {
+	ThingsURL       string `toml:"things_url"`
+	UsersURL        string `toml:"users_url"`
+	ReaderURL       string `toml:"reader_url"`
+	HTTPAdapterURL  string `toml:"http_adapter_url"`
+	BootstrapURL    string `toml:"bootstrap_url"`
+	CertsURL        string `toml:"certs_url"`
+	MsgContentType  string `toml:"msg_content_type"`
+	TLSVerification bool   `toml:"tls_verification"`
+}
+
+type filter struct {
+	Offset          string `toml:"offset"`
+	Limit           string `toml:"limit"`
+	RawOutput       string `toml:"raw_output"`
+	Name            string `toml:"name"`
+	Contact         string `toml:"contact"`
+	Email           string `toml:"email"`
+	Metadata        string `toml:"metadata"`
+	MsgContentType  string `toml:"msg_content_type"`
+	TLSVerification string `toml:"tls_verification"`
+}
+
+type channel struct {
+	Status string `toml:"status"`
+	State  string `toml:"state"`
+	Topic  string `toml:"topic"`
+}
+
 type Config struct {
-	Offset    uint   `toml:"offset"`
-	Limit     uint   `toml:"limit"`
-	Name      string `toml:"name"`
-	RawOutput bool   `toml:"raw_output"`
+	Remotes remotes `toml:"remotes"`
+	Filter  filter  `toml:"filter"`
+	Channel channel `toml:"channel"`
 }
 
-// read - retrieve config from a file.
 func read(file string) (Config, error) {
-	data, err := os.ReadFile(file)
 	c := Config{}
+	data, err := os.Open(file)
 	if err != nil {
-		return c, errors.New(fmt.Sprintf("failed to read config file: %s", err))
+		return c, errors.Wrap(errors.New("Failed to read config file."), err)
 	}
+	defer data.Close()
 
-	if err := toml.Unmarshal(data, &c); err != nil {
-		return Config{}, errors.New(fmt.Sprintf("failed to unmarshal config TOML: %s", err))
+	buf, err := io.ReadAll(data)
+	if err != nil {
+		return c, errors.Wrap(errors.New("Failed to read config file."), err)
+	}
+
+	if err := toml.Unmarshal(buf, &c); err != nil {
+		return Config{}, errors.Wrap(errors.New("Failed to Unmarshall config TOML."), err)
 	}
+
 	return c, nil
 }
 
-func ParseConfig() {
+func ParseConfig(sdkConf mfxsdk.Config) mfxsdk.Config {
 	if ConfigPath == "" {
-		// No config file
-		return
+		return sdkConf
 	}
 
 	if _, err := os.Stat(ConfigPath); os.IsNotExist(err) {
 		errConfigNotFound := errors.Wrap(errors.New("config file was not found"), err)
 		logError(errConfigNotFound)
-		return
+		return sdkConf
 	}
 
 	config, err := read(ConfigPath)
 	if err != nil {
-		log.Fatal(err)
+		logError(errors.Wrap(errors.New("Failed to read config file."), err))
```

Comment:
```suggestion
		logError(errors.Wrap(errors.New("failed to read config file"), err))
```

Reply:
this was previously fatal I would expect it to return the error after which you'd call fatal in main()

Reply:
CLI uses [this](https://github.com/mainflux/mainflux/blob/master/cli/utils.go) to log errors. Should I use a different logger or approach to handle the fatal error?